lib/project.ml

```diff
@@ -4,20 +4,47 @@ type t = Fpath.t
 
 let folder_blacklist = [ "_build"; "_opam"; Fpath.to_string Config.vendor_dir ]
 
+let repo_filename = "repo"
+
+let packages_folder = "packages"
+
+let is_repo_file path =
+  let basename = Fpath.to_string (Fpath.base path) in
+  if String.equal basename repo_filename then Bos.OS.File.exists path
+  else Ok false
```

Comment:
I would suggest checking the file, either with the OPAM API (`OpamFile.Repo.read_opt`) or by just searching for the string `opam-version:` as a poor-mans detection. This should be good enough to not require checking for `"packages"`.

Reply:
`read_opt` will raise, similarly to `read`, I'm still struggling to find where that exception is defined but I don't think it's exposed unfortunately so no good way to deal with this.

Reply:
I'm completely alright with checking for `opam-version:` which I think would handle probably 99,9% of all repos in the wild.

Reply:
I managed to simply use the opam parser for this. It's worth noting than an empty file is a valid repo file so it's good that we didn't implement the check ourselves!